### ISSUE TITLE
[QUERY] How can I get an Upload Stream?

### ISSUE THREAD

**Issue by justinmchase** [Azure/azure-sdk-for-net#35587]
### Library name and version

Azure.Storage.Blobs (all)

### Query/Question

Below is an example snippet of uploading a stream:
```cs
await blob.UploadFromStreamAsync(stream);
```

The problem is that the control in this case is backwards. You first create a stream then you upload, the await causes you to wait for the entire upload to finish before completing, additionally, it implies that you have completely populated the stream before beginning the upload.

What I need is the opposite, I need to get an upload stream, then push into it and then await completion after. For example this:

```cs
var upload = blob.GetUploadStream();
upload.Write(bytes);
upload.Write(bytes);
await upload.Complete();
```

Basically we are dynamically building a PDF document and the size is such that I cannot load it all into memory. What I want is to get the upload stream and have it build the pdf directly into the stream and completely cut out the need for a `MemoryStream`. 

This is definitely possible in nodejs but I'm just not seeing an API to get a write stream from a blob in C#. Is there some equivalent way to do this?

### Environment

dotnet 7, alpine linux, kuberenetes

**Comment by justinmchase** [comment]
Actually I think I found it:

```cs
var upload = await blob.OpenWriteAsync();
```
